pkgs/development/python-modules/greatfet/default.nix

```diff
@@ -1,34 +1,54 @@
 {
-  lib,
-  fetchFromGitHub,
   buildPythonPackage,
-  isPy3k,
+  cmsis-svd,
+  fetchFromGitHub,
   future,
-  pyusb,
   ipython,
+  lib,
+  prompt-toolkit,
+  pyfwup,
   pygreat,
+  pythonOlder,
+  pyusb,
+  setuptools-git-versioning,
+  setuptools,
+  tabulate,
+  tqdm,
 }:
 
 buildPythonPackage rec {
   pname = "greatfet";
   version = "2024.0.1";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "greatscottgadgets";
     repo = "greatfet";
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-AKpaJZJTzMY3IQXLvVnLWh3IHeGp759z6tvaBl28BHQ=";
+    hash = "sha256-AKpaJZJTzMY3IQXLvVnLWh3IHeGp759z6tvaBl28BHQ=";
   };
 
-  disabled = !isPy3k;
+  disabled = pythonOlder "3.8";
+
+  build-system = [
+    setuptools
+    setuptools-git-versioning
```

Comment:
```suggestion
```
Let's patch that out because we set the version manually.

Reply:
It doesn't complain if it's not there. So I guess we can just leave it as is